src/NuGet.Core/NuGet.Protocol/LocalRepositories/LocalPackageSearchResource.cs

```diff
@@ -35,7 +35,7 @@ public LocalPackageSearchResource(FindLocalPackagesResource localResource)
             ILogger log,
             CancellationToken token)
         {
-            return await Task.Run(() =>
+            return await Task.Factory.StartNew(() =>
```

Comment:
Note that this directly contradicts David Fowler's advice: https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#prefer-asyncawait-over-directly-returning-task

He does note that there are perf considerations. The advantage I personally find most compelling is in methods that do some work before calling (and returning) an async method's Task, it will change exception behaviour. `async` methods will always return an `AggregateException`, whereas a non-async method will throw the "real" exception directly, which might break the logic of callers excepection handling expectations (always get an AggregateException).

In this particular case where there's no additional logic, therefore no chance of exception, just calling another Async method and returning immediately, there's little downside to removing the async keyword from the method signature. But I would need convincing to treat "change all `return await` to `return` without `async` on the method signature" to be a rule of thumb to always follow.

Reply:
Sure, I'll try to keep number of machine states down for new code.
But I'll keep as it's now, there're quite many overrides share signature with this method, we can revisit it later if needed.